core/services/ocr2/plugins/ocr2keeper/integration_21_test.go

```diff
@@ -254,6 +257,128 @@ func TestIntegration_KeeperPluginLogUpkeep(t *testing.T) {
+
+		for key, value := range r.Form {
+			t.Logf("MercuryHTTPServe:FormValue: key: %s; value: %s;", key, value)
+		}
+
+		// the streams lookup retries against the remote server 3 times before
+		// returning a result as retryable.
+		// the simulation here should force the streams lookup process to return
+		// retryable 2 times.
+		// the total count of failures should be (upkeepCount * 3 * tryCount)
+		if count <= mercuryFailCount {
+			w.WriteHeader(http.StatusNotFound)
+
+			return
+		}
+
+		// start sending success messages
+		output := `{"chainlinkBlob":"0x0001c38d71fed6c320b90e84b6f559459814d068e2a1700adc931ca9717d4fe70000000000000000000000000000000000000000000000000000000001a80b52b4bf1233f9cb71144a253a1791b202113c4ab4a92fa1b176d684b4959666ff8200000000000000000000000000000000000000000000000000000000000000e000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000000000260000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000001004254432d5553442d415242495452554d2d544553544e4554000000000000000000000000000000000000000000000000000000000000000000000000645570be000000000000000000000000000000000000000000000000000002af2b818dc5000000000000000000000000000000000000000000000000000002af2426faf3000000000000000000000000000000000000000000000000000002af32dc209700000000000000000000000000000000000000000000000000000000012130f8df0a9745bb6ad5e2df605e158ba8ad8a33ef8a0acf9851f0f01668a3a3f2b68600000000000000000000000000000000000000000000000000000000012130f60000000000000000000000000000000000000000000000000000000000000002c4a7958dce105089cf5edb68dad7dcfe8618d7784eb397f97d5a5fade78c11a58275aebda478968e545f7e3657aba9dcbe8d44605e4c6fde3e24edd5e22c94270000000000000000000000000000000000000000000000000000000000000002459c12d33986018a8959566d145225f0c4a4e61a9a3f50361ccff397899314f0018162cf10cd89897635a0bb62a822355bd199d09f4abe76e4d05261bb44733d"}`
+
+		w.WriteHeader(http.StatusOK)
+		_, _ = w.Write([]byte(output))
+	})
+
+	defer mercuryServer.Stop()
+
+	_, err = linkToken.Transfer(linkOwner, upkeepOwner.From, big.NewInt(0).Mul(oneHunEth, big.NewInt(int64(upkeepCount+1))))
+	require.NoError(t, err)
+
+	backend.Commit()
+
+	feeds, err := newFeedLookupUpkeepController(backend, registryOwner)
+	require.NoError(t, err, "no error expected from creating a feed lookup controller")
+
+	// deploy multiple upkeeps that listen to a log emitter and need to be
+	// performed for each log event
+	_ = feeds.DeployUpkeeps(t, backend, upkeepOwner, upkeepCount)
+	_ = feeds.RegisterAndFund(t, registerAndFund(registry, registryOwner, backend, linkToken))
+	_ = feeds.EnableMercury(t, backend, registry, registryOwner)
+	_ = feeds.VerifyEnv(t, backend, registry, registryOwner)
+
+	// using sleep here is not the right way to go, but it currently work on
+	// a local machine. this does have issues in CI
+	time.Sleep(10 * time.Second)
```

Comment:
curious, what is the issue in CI?

Reply:
Let's avoid the sleep as it might be flaky, can't we wait for something to happen?

Reply:
Actually I'll try removing this. Amir did some updates where this timer wasn't needed.